x/spec/types/spec.go

```diff
@@ -7,6 +7,97 @@ import (
 
 const minCU = 1
 
+type getterFunc func(index string) (Spec, bool)
+
+// ExpandSpec expands the "imports" field of the spec -if needed, recursively-
+// and adds to the current Spec those additional APIs from the imported Spec(s).
+// It takes as argument a getter function to fetch imported Spec(s) by name from
+// the keeper store, on demand. (Can't import the keeper module due to circular
+// dependencies)
+func (spec *Spec) ExpandSpec(getSpec getterFunc) (map[string]string, error) {
+	depends := map[string]bool{spec.Index: true}
+	details, err := spec.doExpandSpec(getSpec, depends, spec.Index)
+	return map[string]string{"spec": spec.Name, "imports": details}, err
+}
+
+// doExpandSpec performs the actual work and recusion for ExpandSpec above.
+func (spec *Spec) doExpandSpec(getSpec getterFunc, depends map[string]bool, details string) (string, error) {
+	if len(spec.Imports) == 0 {
+		return details, nil
+	}
+
+	var parents []Spec
+
+	// visual markers when import deepens
+	details += "->["
+
+	// recursion to get all parent specs
+	comma := ""
+	for _, index := range spec.Imports {
+		imported, found := getSpec(index)
+		// import of unknown Spec not allwoed
+		if !found {
+			details += fmt.Sprintf("%s%s(unknown)", comma, index)
+			return details, fmt.Errorf("imported spec unknown: %s", index)
+		}
+		// import of disabled Spec not allowed
+		if !imported.Enabled {
+			details += fmt.Sprintf("%s%s(disabled)", comma, index)
```

Comment:
importing disabled specs should be allowed, in case we want to inherit from some base spec that doesn't hold on it's own

Reply:
ok